backend/internal/cmds/import.go

```diff
@@ -0,0 +1,371 @@
+		return fmt.Errorf(
+			"import file is for %s, but current DB provider is %s",
+			dbData.Provider,
+			common.EnvConfig.DbProvider,
+		)
+	}
+
+	// Connect DB and reset schema
+	db, err := bootstrap.NewDatabase()
+	if err != nil {
+		return err
+	}
+	if err := resetSchema(db); err != nil {
+		return err
+	}
+
+	if err := runMigrations(db, dbData.Version); err != nil {
+		return err
+	}
+
+	if err := insertData(db, dbData); err != nil {
+		return err
+	}
+
+	if err := processZipFiles(r.File); err != nil {
+		return err
+	}
+
+	fmt.Println("Import completed successfully.")
+	return nil
+}
+
+func askForConfirmation() bool {
+	fmt.Println("WARNING: This feature is experimental and may not work correctly. Please create a backup before proceeding and report any issues you encounter.")
+	fmt.Println()
+	fmt.Println("WARNING: Import will erase all existing data at the following locations:")
+	fmt.Printf("Database:      %s\n", absolutePathOrOriginal(common.EnvConfig.DbConnectionString))
+	fmt.Printf("Uploads Path:  %s\n", absolutePathOrOriginal(common.EnvConfig.UploadPath))
+
+	fmt.Printf("Keys Path:     %s\n", absolutePathOrOriginal(common.EnvConfig.KeysPath))
+
+	ok, err := utils.PromptForConfirmation("Do you want to continue?")
+	if err != nil {
+		panic(err)
+	}
+	return ok
+
+}
+
+// processZipDatabaseJson extracts database.json from the ZIP archive
+func processZipDatabaseJson(files []*zip.File) (databaseJson, error) {
+	var dbData databaseJson
+
+	for _, f := range files {
+		if f.Name == "database.json" {
+			if err := readDatabaseJSON(f, &dbData); err != nil {
+				return dbData, err
+			}
+			return dbData, nil
+		}
+	}
+	return dbData, errors.New("database.json not found in the ZIP file")
+}
+
+// processZipFiles extracts uploads/ and keys/ from the ZIP archive
+func processZipFiles(files []*zip.File) error {
+	for _, f := range files {
+		switch {
+		case strings.HasPrefix(f.Name, "uploads/"):
+			if err := extractIntoBase(f, common.EnvConfig.UploadPath, "uploads/"); err != nil {
+				return fmt.Errorf("failed to extract uploads: %w", err)
+			}
+
+		case strings.HasPrefix(f.Name, "keys/"):
+			if err := extractIntoBase(f, common.EnvConfig.KeysPath, "keys/"); err != nil {
+				return fmt.Errorf("failed to extract keys: %w", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// readDatabaseJSON parses database.json from the ZIP file
+func readDatabaseJSON(f *zip.File, dbData *databaseJson) error {
+	rc, err := f.Open()
+	if err != nil {
+		return fmt.Errorf("failed to open database.json: %w", err)
+	}
+	defer rc.Close()
+
+	if err := json.NewDecoder(rc).Decode(dbData); err != nil {
+		return fmt.Errorf("failed to decode database.json: %w", err)
+	}
+	return nil
+}
+
+// resetSchema clears the DB schema depending on the provider
+func resetSchema(db *gorm.DB) error {
+	switch common.EnvConfig.DbProvider {
+	case common.DbProviderPostgres:
+		return db.Exec(`DROP SCHEMA public CASCADE; CREATE SCHEMA public;`).Error
```

Comment:
I don't think it's safe to assume the schema name is always "public", isn't it? IIRC the connection string could define a different schema name.